### PR TITLE
Dịch "rnn-scratch_vn" - Phần 4

### LINKED ISSUE #1546: Dịch "rnn-scratch_vn" - Phần 4
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_recurrent-neural-networks/rnn-scratch_vn.md#L281-L334

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by tranyenthy**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #1546 

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [ ] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_recurrent-neural-networks/rnn-scratch_vn.md [hunk 1]
```diff
@@ -286,7 +286,9 @@ This is both a curse and a blessing.
 A curse since it limits the speed of making progress, whereas a blessing since it limits the extent to which things can go wrong if we move in the wrong direction.
 -->
 
-*dịch đoạn phía trên*
+Trong trường hợp này, ta có thể giả định một cách an toàn rằng nếu chúng tôi cập nhật vector trọng số theo $\eta \cdot \mathbf{g}_t$, ta sẽ không quan sát thấy sự thay đổi nhiều hơn $L \eta \|\mathbf{g}_t\|$.
+Đây là cả một lời nguyền và một phước lành.
+Nó là một lời nguyền vì nó giới hạn tốc độ tiến bộ, trong khi một phước lành vì nó giới hạn mức độ mà mọi thứ có thể đi sai nếu chúng ta đi sai hướng.
```
Comment (quangnhat185) [inline chapter_recurrent-neural-networks/rnn-scratch_vn.md:291]:
```suggestion
Trong trường hợp này, ta có thể giả định một cách an toàn rằng bằng cách cập nhật vector trọng số theo $\eta \cdot \mathbf{g}_t$, sự thay đổi sẽ không lớn hơn $L \eta \|\mathbf{g}_t\|$.
Đây là cả một lời nguyền và một phước lành.
Một lời nguyền là vì nó giới hạn tốc độ tiến bộ,  một phước lành là bởi nó hạn chế mức độ sai lệnh trong trường hợp chúng ta đi sai hướng.
```

File: chapter_recurrent-neural-networks/rnn-scratch_vn.md [hunk 2]
```diff
@@ -296,7 +298,11 @@ In this case such an approach may appear entirely unwarranted.
 One alternative is to clip the gradients by projecting them back to a ball of a given radius, say $\theta$ via
 -->
 
-*dịch đoạn phía trên*
+Đôi khi gradient có thể khá lớn và thuật toán tối ưu có thể không hội tụ.
+Ta có thể giải quyết vấn đề này bằng cách giảm tốc độ học $\eta$ hoặc bằng một số thủ thuật bậc cao khác.
+Nhưng điều gì sẽ xảy ra nếu chúng ta hiếm khi nhận được độ dốc lớn?
```
Comment (quangnhat185) [inline chapter_recurrent-neural-networks/rnn-scratch_vn.md:303]:
```suggestion
Nhưng điều gì sẽ xảy ra nếu chúng ta hiếm khi nhận được gradient lớn?
```

File: chapter_recurrent-neural-networks/rnn-scratch_vn.md [hunk 3]
```diff
@@ -296,7 +298,11 @@ In this case such an approach may appear entirely unwarranted.
...
+Trong trường hợp này, cách tiếp cận như vậy có thể xuất hiện hoàn toàn không chính đáng.
+Một cách khác là cắt bớt các gradient bằng cách chiếu chúng trở lại một quả bóng có bán kính nhất định, giả sử $\theta$ thông qua
```
Comment (quangnhat185) [inline chapter_recurrent-neural-networks/rnn-scratch_vn.md:305]:
```suggestion
Trong trường hợp này, cách tiếp cận như vậy không được bảo đảm hoàn toàn.
Một cách khác là cắt bớt các gradient bằng cách chiếu chúng trở lại một quả cầu với bán kính $\theta$ thông qua:
```

File: chapter_recurrent-neural-networks/rnn-scratch_vn.md [hunk 4]
```diff
@@ -308,14 +314,19 @@ Gradient clipping provides a quick fix to the gradient exploding.
 While it does not entirely solve the problem, it is one of the many techniques to alleviate it.
 -->
 
-*dịch đoạn phía trên*
+Bằng cách làm như vậy, chúng tôi biết rằng định mức độ dốc không bao giờ vượt quá $\theta$ và độ dốc được cập nhật hoàn toàn phù hợp với hướng ban đầu $\mathbf{g}$.
+Nó cũng có tác dụng phụ mong muốn là hạn chế ảnh hưởng của bất kỳ xe buýt nhỏ nào (và bên trong nó là bất kỳ mẫu nào) có thể tác động lên các vector trọng số.
+Điều này mang lại một độ mạnh mẽ nhất định cho mô hình.
+Cắt gradient cung cấp một sửa chữa nhanh chóng cho gradient phát nổ.
+Mặc dù nó không hoàn toàn giải quyết vấn đề, nhưng nó là một trong nhiều kỹ thuật để giảm bớt nó.
```
Comment (quangnhat185) [inline chapter_recurrent-neural-networks/rnn-scratch_vn.md:321]:
```suggestion
Bằng cách làm như vậy, ta biết rằng chuẩn độ dốc không bao giờ vượt quá $\theta$ và độ dốc được cập nhật hoàn toàn phù hợp với hướng ban đầu $\mathbf{g}$.
Nó cũng có tác dụng phụ tích cực là hạn chế ảnh hưởng của bất kỳ minibatch nào (và bên trong nó là bất kỳ mẫu nào) có thể tác động lên các vector trọng số.
Điều này mang lại một độ mạnh mẽ nhất định cho mô hình.
Gọt gradient là một phương án sửa chữa nhanh chóng cho vấn đề gradient phát nổ.
Mặc dù nó không hoàn toàn giải quyết vấn đề, nhưng là một trong nhiều kỹ thuật để giảm bớt vấn đề đó.
```

File: chapter_recurrent-neural-networks/rnn-scratch_vn.md [hunk 5]
```diff
@@ -308,14 +314,19 @@ Gradient clipping provides a quick fix to the gradient exploding.
...
 
 <!--
 Below we define a function to clip the gradients of a model that is either a `RNNModelScratch` instance or a Gluon model.
 Also note that we compute the gradient norm over all parameters.
 -->
 
-*dịch đoạn phía trên*
+Dưới đây, chúng tôi định nghĩa một hàm để cắt các gradient của một mô hình là một thể hiện `RNNModelScratch` hoặc một mô hình Gluon.
+Cũng lưu ý rằng chúng tôi tính toán trung bình gradient trên tất cả các tham số.
```
Comment (quangnhat185) [inline chapter_recurrent-neural-networks/rnn-scratch_vn.md:329]:
```suggestion
Dưới đây, chúng tôi định nghĩa một hàm để gọt các gradient của mô hình: hoặc là một thực thể `RNNModelScratch` hoặc là một mô hình Gluon.
Cũng lưu ý rằng chúng tôi tính toán trung bình gradient trên tất cả các tham số.
```
